packages/api-bindings/src/lens/utils/CollectModuleSettings.ts

```diff
@@ -141,64 +190,89 @@ export function resolveCollectPolicy(collectable: PrimaryPublication): CollectPo
 
   if (!module) return null;
 
-  const fee = buildCollectFee(module);
   const shared = {
     followerOnly: module.followerOnly,
     contract: module.contract,
   };
 
   switch (module.__typename) {
     case 'LegacyAaveFeeCollectModuleSettings':
-    case 'LegacyERC4626FeeCollectModuleSettings': {
+    case 'LegacyERC4626FeeCollectModuleSettings':
       return {
         ...shared,
         collectNft: null,
         collectLimit: module.collectLimit,
         endsAt: module.endsAt,
-        fee,
+        fee: buildCollectFee(module),
       };
-    }
+
     case 'LegacyLimitedFeeCollectModuleSettings':
-    case 'LegacyLimitedTimedFeeCollectModuleSettings': {
+    case 'LegacyLimitedTimedFeeCollectModuleSettings':
       return {
         ...shared,
         collectNft: module.collectNft,
         collectLimit: module.collectLimit,
         endsAt: null,
-        fee,
+        fee: buildCollectFee(module),
       };
-    }
+
     case 'LegacyFeeCollectModuleSettings':
-    case 'LegacyTimedFeeCollectModuleSettings': {
+    case 'LegacyTimedFeeCollectModuleSettings':
       return {
         ...shared,
         collectNft: module.collectNft,
         collectLimit: null,
         endsAt: null,
-        fee,
+        fee: buildCollectFee(module),
       };
-    }
+
     case 'LegacyFreeCollectModuleSettings':
       return {
         ...shared,
         collectNft: module.collectNft,
         collectLimit: null,
         endsAt: null,
       };
+
     case 'LegacyMultirecipientFeeCollectModuleSettings':
     case 'MultirecipientFeeCollectOpenActionSettings':
+      return {
+        ...shared,
+        collectNft: module.collectNft,
+        collectLimit: module.collectLimit,
+        endsAt: module.endsAt,
+        fee: buildCollectFee(module),
+      };
+
     case 'LegacySimpleCollectModuleSettings':
-    case 'SimpleCollectOpenActionSettings': {
+    case 'SimpleCollectOpenActionSettings':
       return {
         ...shared,
         collectNft: module.collectNft,
         collectLimit: module.collectLimit,
         endsAt: module.endsAt,
-        fee,
+        fee: buildCollectFee(module),
       };
-    }
 
-    default:
-      return null;
+    case 'ProtocolSharedRevenueCollectOpenActionSettings': {
+      const fee = buildCollectFee(module);
+
+      if (fee) {
+        return {
+          ...shared,
+          collectNft: module.collectNft,
+          collectLimit: module.collectLimit,
+          endsAt: module.endsAt,
+          fee: fee,
+        };
+      }
+      return {
+        ...shared,
+        collectNft: module.collectNft,
+        collectLimit: module.collectLimit,
+        endsAt: module.endsAt,
+        creatorClient: module.creatorClient ?? undefined,
```

Comment:
so creatorClient is defined only if no fee?

Reply:
correct